packages/lesswrong/lib/ultraFeed/UltraFeedRepo.ts

```diff
@@ -0,0 +1,413 @@
+import { FeedItemSourceType, FeedSpotlight, PreDisplayFeedComment, PreDisplayFeedCommentThread, FeedFullPost, FeedPostMetaInfo, FeedCommentsThread, FeedCommentFromDb } from '../../components/ultraFeed/ultraFeedTypes';
+import Comments from '../../server/collections/comments/collection';
+import AbstractRepo from '../../server/repos/AbstractRepo';
+import { recordPerfMetrics } from '../../server/repos/perfMetricWrapper';
+import { prioritizeThreads, prepareCommentThreadForResolver  } from './ultraFeedThreadHelpers';
+import { recombeeApi, recombeeRequestHelpers } from '../../server/recombee/client';
+import { HybridRecombeeConfiguration } from '../../lib/collections/users/recommendationSettings';
+import { aboutPostIdSetting } from '@/lib/instanceSettings';
+
+const COMMENT_LOOKBACK_WINDOW = '90 days';
+
+export interface ThreadStatistics {
+  commentCount: number;
+  maxKarma: number;
+  sumKarma: number;
+  sumKarmaSquared: number;
+  averageKarma: number;
+  averageTop3Comments: number;
+}
+
+class UltraFeedRepo extends AbstractRepo<"Comments"> {
+  constructor() {
+    super(Comments);
+  }
+
+  public async getUltraFeedPostThreads(
```

Comment:
Methods are public by default, you don't need the modifier.  Also, this doesn't look like it's making a database query at all, so I'm not sure why it's on the repo